src/barchart_stacked/BarChartStacked.js

```diff
@@ -0,0 +1,347 @@
+    this.g.append('text').attr("class", "axis-x-legend").attr("x", (-this.margin.left/3)).attr("y", this.height + 9).attr("dy", "0.71em").attr("text-anchor", "end").text(this.xAxisProp);
+  }
+
+  build() {
+    this.setScales();
+
+    this.g
+      .select(".axis-x")
+      .attr("transform", `translate(0 ${this.height})`)
+      .call(this.xAxis.bind(this));
+
+    this.g
+      .select(".axis-y")
+      .call(this.yAxis.bind(this));
+
+    this.g
+      .selectAll(".bar-stacked-group")
+      .data(stack().keys(this.columns)(this.data))
+      .join("g")
+        .attr("class", "bar-stacked-group")
+        .attr("id", ({ key }) => key)
+        .attr("fill", ({ key }) => this.scaleColor(key))
+      .selectAll("rect")
+      .data(d => d)
+      .join("rect")
+        .attr("class", "bar-stacked-rect")
+        .attr("x", d => this.scaleX(d.data[this.xAxisProp]))
+        .attr("y", ([y1, y2]) => Math.min(this.scaleY(y1), this.scaleY(y2)))
+        .attr("width", this.scaleX.bandwidth())
+        .attr('height', ([y1, y2]) => Math.abs(this.scaleY(y1) - this.scaleY(y2)))
+      .on("touchmove", e => e.preventDefault())
+      .on("pointermove", this.onPointerMove.bind(this))
+      .on("pointerout", this.onPointerOut.bind(this))
+      .attr("cursor", "pointer")
+      .on("click", (...e) => this.onClick(...e));
+
+    if(this.showLegend) {
+      this.buildLegends()
+    }
+
+    if(this.extraLegends.length) {
+      this.buildExtraAxis()
+    }
+  }
+
+  buildLegends() {
+    const positionLengedGroupX = this.orientationLegend === 'left' ? 0 : (this.width + this.margin.left);
+    const positionLengedLabelX = this.orientationLegend === 'left' ? 24 : (positionLengedGroupX + 24);
+    this.svg
+      .selectAll(".bar-stack-label")
+      .data(stack().keys(this.columns)(this.data))
+      .join(
+        enter => {
+          const g = enter
+            .append("g")
+            .attr("class", "bar-stack-label")
+            .attr("fill", ({ key }) => this.scaleColor(key))
+            .attr("transform", (d, i) => `translate(10, ${i * 24})`)
+          g.append("rect")
+            .attr("x", positionLengedGroupX)
+            .attr("y", (d, i) => `${this.margin.top + (i * 3)}`)
+            .attr("class", "bar-stack-label-rect")
+            .attr("width", 16)
+            .attr("height", 16)
+          g.append("text")
+            .attr("class", "bar-stacked-legend-text")
+            .attr("x", positionLengedLabelX)
+            .attr("y", (d, i) => `${this.margin.top + (i * 3) + 14}`)
+            .text(({ key }) => key);
+
+          return g;
+        },
+        update => update,
+        exit => exit.remove()
+      )
+      .on("pointermove", function(i, d, _) {
+          const { key } = d
+
+          selectAll('.bar-stacked-rect')
+            .style("opacity", .2)
+
+          selectAll(`#${key} .bar-stacked-rect`)
+            .style("opacity", 1)
```

Comment:
This is about redoing the wheel, penalizing the performance:

_key_ is a dataset property, that means, it could be filtered. Therefore:
```suggestion
          const rects = selectAll('.bar-stacked-rect')
          
          rects.filter(({ key: k }) => k !== key)
            .style("opacity", .2)

          rects.filter(({ key: k }) => k === key)
            .style("opacity", 1)
```

You save a DOM-tree seek foreach event